drizzlepac/runastrodriz.py

```diff
@@ -2133,6 +2201,52 @@ def _analyze_exposure(filename):
     return process_exposure
 
 
+def _update_idcscale(filename):
+    """
+    Update IDCSCALE if a headerlet does not include it.
+
+    Parameters
+    ----------
+    filename : str or fits.HDUList
+        The science file to be updated.
+        If HDUList it must be opened in "update" mode.
+
+    """
+
+    if isinstance(filename, str):
+        hdul = fits.open(filename, mode='update')
+    else:
+        # Assume it's an HDUList object in "update" mode.
+        hdul = filename
+    num_sci = fileutil.countExtn(hdul)
+    # get IDCTAB name
+    itabroot = hdul[0].header['idctab'].split('$')[1].split('_')[0]
+    fhdu_idscale = None
+    # pull a value from one of the other headerlet extensions
+    extvers = headerlet.get_headerlet_kw_names(hdul, kw='extver')
+    for sciext in range(num_sci):
+        for extn in extvers:
+            if itabroot in hdul[('HDRLET', extn)].header['wcsname']:
+                _hdrlet = hdul[('HDRLET', extn)].headerlet
+                if 'idcscale' in _hdrlet[('SIPWCS', sciext + 1)].header:
+                    fhdu_idscale = _hdrlet[('SIPWCS', sciext + 1)].header['idcscale']
+                    break
+        if fhdu_idscale is None:
+            cd11 = hdul[('sci', sciext + 1)].header['CD1_1']
+            cd21 = hdul[('sci', sciext + 1)].header['CD2_1']
+            fhdu_idscale = round(np.sqrt(np.power(cd11, 2) + np.power(cd21, 2)) * 3600., 3)
```

Comment:
What is the reason for rounding  idcscale? I don't know if this image will be used in sky subtraction, but if it does - it looks like idcscale is used there to scale computed sky values. Also, wouldn't `np.sqrt(cd11**2 + cd21**2)` do the same? It seems that using `numpy`'s `power` will just add overhead.

Reply:
Agreed on using `np.power`. I basically copied the expression that was used in the apriori fix to avoid getting any differences in results. But I agree it can be cleaned up.

Reply:
_If_ these files will ever be run through `astrodrizzle`, the `idcscale` value may be used in computations and accuracy matters. There is no reason to use `round` on this value anywhere. Maybe it shouldn't be done in this PR but these calls to `round` should be removed.

Reply:
FIled #2051